catalog/transformer/catalog-transformer-csv/src/test/java/ddf/catalog/transformer/csv/CsvQueryResponseTransformerTest.java

```diff
@@ -0,0 +1,236 @@
+        Scanner scanner = new Scanner(bc.getInputStream());
+        scanner.useDelimiter("\\n|\\r|,");
+
+        //OBJECT types, BINARY types and hidden attributes will be filtered out
+        String[] expectedHeaders = {"attribute4", "column2", "column1"};
+        validate(scanner, expectedHeaders);
+
+        //The scanner will split "value,4" into two tokens even though the CSVPrinter will
+        //handle it correctly.
+        String[] expectedValues = {"", "\"value", "4\"", "101", "value1"};
+
+        for (int i = 0; i < METACARD_COUNT; i++) {
+            validate(scanner, expectedValues);
+        }
+
+        //final new line causes an extra "" value at end of file
+        assertThat(scanner.hasNext(), is(true));
+        assertThat(scanner.next(), is(""));
+        assertThat(scanner.hasNext(), is(false));
+    }
+
+    private void validate(Scanner scanner, String[] expectedValues) {
+        for (int i = 0; i < expectedValues.length; i++) {
+            assertThat(scanner.hasNext(), is(true));
+            assertThat(scanner.next(), is(expectedValues[i]));
+        }
+    }
+
+    private ArrayList<String> buildList(String[] entries) {
+        ArrayList<String> arrayList = new ArrayList<>();
+        arrayList.addAll(Arrays.asList(entries));
+        return arrayList;
+    }
+
+    private HashMap<String, String> buildMap(String[][] entries) {
+        HashMap<String, String> hashMap = new HashMap<>();
+
+        for (String[] entry : entries) {
+            hashMap.put(entry[0], entry[1]);
+        }
+
+        return hashMap;
+    }
+
+    private HashSet<String> buildSet(String[] attributes) {
+        HashSet<String> hashSet = new HashSet<>();
+        hashSet.addAll(Arrays.asList(attributes));
+        return hashSet;
+    }
+
+    private void buildMetacardDataMap() {
+        for (Object[] entry : ATTRIBUTE_DATA) {
+            String attributeName = entry[0].toString();
+            AttributeType attributeType = (AttributeType) entry[2];
+            Serializable attributeValue = (Serializable) entry[1];
+            Attribute attribute = buildAttribute(attributeName, attributeValue);
+            METACARD_DATA_MAP.put(attributeName, attribute);
+            ATTRIBUTE_DESCRIPTOR_LIST.add(buildAttributeDescriptor(attributeName, attributeType));
+        }
+    }
+
+    private SourceResponse buildSourceResponse() {
+        SourceResponse sourceResponse = mock(SourceResponse.class);
+        when(sourceResponse.getResults()).thenReturn(RESULT_LIST);
+        return sourceResponse;
+    }
+
+    private void buildResultList() {
+        RESULT_LIST.clear();
+
+        for (int i = 0; i < METACARD_COUNT; i++) {
+            Result result = mock(Result.class);
+            Metacard metacard = buildMetacard(i);
+            when(result.getMetacard()).thenReturn(metacard);
+            RESULT_LIST.add(result);
+        }
+    }
+
+    private Metacard buildMetacard(final int id) {
```

Comment:
Codacity is complaining for you to remove this since it is never used, Though not sure why it didn't comment on the PR itself.